save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/PluginConfig.kt

```diff
@@ -48,30 +46,24 @@ data class GeneralConfig(
     val excludedTests: String? = null,
     val includedTests: String? = null,
     val ignoreSaveComments: Boolean? = null
-) : PluginConfig<GeneralConfig> {
-    @Suppress("TYPE_ALIAS")
-    override fun mergeConfigInto(childConfig: MutableList<PluginConfig<*>>) {
-        val childGeneralConfig = childConfig.filterIsInstance<GeneralConfig>().firstOrNull()
-        val newChildGeneralConfig = childGeneralConfig?.mergePluginConfig(this) ?: this
-        // Now we update child config in place
-        childGeneralConfig?.let {
-            childConfig.set(childConfig.indexOf(childGeneralConfig), newChildGeneralConfig)
-        } ?: childConfig.add(newChildGeneralConfig)
-    }
+) : PluginConfig {
+    override val type = TestConfigSections.GENERAL
 
-    override fun mergePluginConfig(parentConfig: GeneralConfig): GeneralConfig {
-        val mergedTag = parentConfig.tags?.let {
-            val parentTags = parentConfig.tags.split(", ")
+    override fun mergeWith(otherConfig: PluginConfig): PluginConfig {
+        val other = otherConfig as GeneralConfig
+        val mergedTag = other.tags.let {
+            val parentTags = other.tags.split(", ")
             val childTags = this.tags.split(", ")
             parentTags.union(childTags).joinToString(", ")
-        } ?: this.tags
+        }
+
         return GeneralConfig(
             mergedTag,
-            this.description ?: parentConfig.description,
-            this.suiteName ?: parentConfig.suiteName,
-            this.excludedTests ?: parentConfig.excludedTests,
-            this.includedTests ?: parentConfig.includedTests,
-            this.ignoreSaveComments ?: parentConfig.ignoreSaveComments
+            this.description,
```

Comment:
I thought, that we decide to check all fields for nullability in aim of consistency

Reply:
fixed for static analysis in idea